### ISSUE TITLE
Invalid XAML

### ISSUE THREAD

**Issue by SnackSBR** [artem-zinnatullin/jet-image-loader#28]
I'm trying to implement jet on my project but i have this error:

A first chance exception of type 'System.Windows.Markup.XamlParseException' occurred in System.Windows.ni.dll

Additional information: Cannot create instance of type 'Cartao_de_Visitas.Classes.JetImageLoaderConverter' [Line: 12 Position: 44]

And my page have this area as invalid XAML

```
                <DataTemplate>
                    <Grid Margin="12 0 12 0">
                        <Image Height="200" Source="{Binding ImageUrl, Converter={StaticResource JetImageLoaderConverter}}"/>
                    </Grid>
                </DataTemplate>
```

**Comment by artem-zinnatullin** [comment]
Did you declare `JetImageLoaderConverter` in the `App.xaml`?

You can check `README.md` and Sample app's `App.xaml`

**Comment by SnackSBR** [comment]
xmlns:app="clr-namespace:Cartao_de_Visitas.Classes"

< app:JetImageLoaderConverter x:Key="JetImageLoaderConverter" />

**Comment by artem-zinnatullin** [comment]
From `README.md`

```
<Application
    ...
    xmlns:myApp="clr-namespace:MyApp.NamespaceWithJetImageLoaderConverter>
<Application.Resources>
    <myApp:MyAppJetImageLoaderConverter x:Key="JetImageLoaderConverter"/>
</Application.Resources>
...
```

What is the name of your extended `JetImageLoaderConverter` class, are you sure that you declared it in the `App.xaml` correctly?

**Comment by SnackSBR** [comment]
http://pastebin.com/hiYB1PXJ
http://pastebin.com/bWgJELES
http://pastebin.com/bzJRKtQb

**Comment by artem-zinnatullin** [comment]
Looks valid, hmm... Are you building WP7 or WP8 project?

**Comment by SnackSBR** [comment]
wp8 project

**Comment by artem-zinnatullin** [comment]
Very strange.. Can you please clone repository and try to build JetImageLoader Sample project, may be something wrong with your OS config or something same?

**Comment by SnackSBR** [comment]
sample project runs just fine (already tried). very, very strange.

**Comment by artem-zinnatullin** [comment]
Yeah :( I don't know what to suggest for you, may be clean and rebuild of the project could help?

I can suggest to try to change some parts of `JetImageLoader` impementation, config, try to play with some parameters of `JetImageLoader` config as the error is `Cannot create instance`

**Comment by SnackSBR** [comment]
Tried to clean and rebuild too.
I think its strange because the only difference from sample for mine is the converter name and with my converter name i get the invalid xaml.
I'm missing something on the page apart from
http://pastebin.com/pVyfJD4e
or this is the only xaml modification needed?

**Comment by artem-zinnatullin** [comment]
Hmm, no, `HorizontalContentAligment` is not mandatory of course.

That is super strange I think, because I got two projects with different `JetImageLoader` names and configurations..

May be you should try to re-add `JetImageLoader` exactly as in the Sample app? I really don't know how to help you :(

**Comment by SnackSBR** [comment]
nugget version is the same from jet-image-loader-master/Releases ?

**Comment by artem-zinnatullin** [comment]
No, NuGet version is not so new as master branch, I can push new version but only tommorow, because I am on the Mac right now, I need Windows PC for that

I would recommend you to try add JetImageLoader as sources and if it works, then NuGet updated version should help you

**Comment by SnackSBR** [comment]
Tried and I don't get invalid XAML but still have
A first chance exception of type 'System.Windows.Markup.XamlParseException' occurred in System.Windows.ni.dll

Additional information: Cannot create instance of type 'Cartao_de_Visitas.Classes.JetImageLoaderConverter' [Line: 12 Position: 44]

I'll keep trying. Thanks

**Comment by artem-zinnatullin** [comment]
That's very sad :( If I understand how to help you, I'll comment here

**Comment by shobankr** [comment]
I have the same problem :( tried three different new projects and same error. Sample project works fine. Tried the Nuget package, DLL from sample project and all give the same error.

**Comment by shobankr** [comment]
Okay I found a fix. instead of adding the name space and resource separately trying the following

< app:JetImageLoaderConverter x:Key="JetImageLoaderConverter"  xmlns:app="clr-namespace:Cartao_de_Visitas.Classes"/>

**Comment by artem-zinnatullin** [comment]
Awesome! Thank you

---

Best regards, Artem Zinnatullin

**Comment by SnackSBR** [comment]
Thanks shobankr, i will try later.

**Comment by sgraf812** [comment]
For me it helped to make the class public to it can be found by the runtime loader.
This required for every value converter.
